test/bigml/test/api/source.clj

```diff
@@ -8,11 +8,13 @@
   (:use clojure.test))
 
 (defn- create-and-test [artifact & params]
-  (let [initial (api/get-final (apply source/create artifact params))
+  (let [{:keys [username api_key dev_mode] :as sticky-pars} params
```

Comment:
> On 14 Jan 2016, at 17:45, jimshur <notifications@github.com> wrote:
> 
> In test/bigml/test/api/source.clj:
> 
> > @@ -8,11 +8,13 @@
> >    (:use clojure.test))
> >  
> >  (defn- create-and-test [artifact & params]
> > -  (let [initial (api/get-final (apply source/create artifact params))
> > +  (let [{:keys [username api_key dev_mode] :as sticky-pars} params
> 
> Well, since, as Adam pointed out, you're not using any of the variables in the destructuring, probably just as easy to say
> 
> (let [sticky-pars (select-keys params [:username :api_key :dev_mode])]

Thanks.
I had already tried to use `select-keys`, but `params` is not a map, so that fails:
    clojure.lang.ArraySeq cannot be cast to java.util.Map

I could convert the sequence of `params` to a map, then `select-keys`, but I am not sure it is worth the effort (although `params` should seldom contain many items).

Am I missing anything?

> 
> —
> Reply to this email directly or view it on GitHub.
>